compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs

```diff
@@ -5178,6 +5178,34 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
             _ => {}
         }
     }
+
+    pub fn suggest_remove_brackets_from_range(
+        &self,
+        err: &mut Diag<'_>,
+        main_trait_predicate: ty::PolyTraitPredicate<'tcx>,
+        leaf_trait_predicate: ty::PolyTraitPredicate<'tcx>,
+        span: Span,
+    ) {
+        if self.tcx.is_diagnostic_item(sym::RangeBounds, leaf_trait_predicate.def_id())
+            && let ty::Array(inner_type, arr_size) =
+                main_trait_predicate.self_ty().skip_binder().kind()
+            && let ty::Adt(adt_def, _) = inner_type.kind()
+            && self.tcx.is_range(adt_def.did())
```

Comment:
I just realized, we already have something that describes whether a type is a valid range: the `RangeBounds` trait.

So maybe we can use something like this instead:
```suggestion
            // NOTE: Assumes that arrays do not implement `RangeBounds`.
            && let Some(range_bounds) = self.tcx.get_diagnostic_item(sym::RangeBounds)
            && self.type_implements_trait(range_bounds, [inner_type], param_env).must_apply_modulo_regions()
```

And then we can get rid of the `is_range` function?